# Individual Project 5
<p>Author: Zane Brown<br>Version: 1.0<br>DS-160-01<br>Spring'22<p>

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
data=pd.read_csv('Self Awareness Dataset.csv')

In [3]:
data.head()

Date       Day  Sleep Hour  Exercise Hour  Study Hour  StudyDS Hour  \
0  14-Jan    Friday           6              3           2             1   
1  15-Jan  Saturday           7              2           3             2   
2  16-Jan    Sunday           5              1           3             1   
3  17-Jan    Monday           7              3           3             2   
4  18-Jan   Tuesday           8              3           2             1   

   Work Hour  End of the day Feeling (0 to 10)  Unnamed: 8  Unnamed: 9  \
0          6                                 8         NaN         NaN   
1          4                                 4         NaN         NaN   
2          0                                 6         NaN         NaN   
3          0                                 5         NaN         NaN   
4          0                                 6         NaN         NaN   

   Unnamed: 10  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4          NaN

In [4]:
del data['Unnamed: 8']
del data['Unnamed: 9']
del data['Unnamed: 10']

In [5]:
data.head()

Date       Day  Sleep Hour  Exercise Hour  Study Hour  StudyDS Hour  \
0  14-Jan    Friday           6              3           2             1   
1  15-Jan  Saturday           7              2           3             2   
2  16-Jan    Sunday           5              1           3             1   
3  17-Jan    Monday           7              3           3             2   
4  18-Jan   Tuesday           8              3           2             1   

   Work Hour  End of the day Feeling (0 to 10)  
0          6                                 8  
1          4                                 4  
2          0                                 6  
3          0                                 5  
4          0                                 6

In [6]:
data.isnull().sum()

Date                                0
Day                                 0
Sleep Hour                          0
Exercise Hour                       0
Study Hour                          0
StudyDS Hour                        0
Work Hour                           0
End of the day Feeling (0 to 10)    0
dtype: int64

In [7]:
data.columns

Index(['Date', 'Day', 'Sleep Hour', 'Exercise Hour', 'Study Hour',
       'StudyDS Hour', 'Work Hour', 'End of the day Feeling (0 to 10)'],
      dtype='object')

In [8]:
X=data[['Day','Sleep Hour','Exercise Hour','Study Hour','StudyDS Hour','Work Hour']]
y=data[['End of the day Feeling (0 to 10)']]

In [9]:
X.head()

Day  Sleep Hour  Exercise Hour  Study Hour  StudyDS Hour  Work Hour
0    Friday           6              3           2             1          6
1  Saturday           7              2           3             2          4
2    Sunday           5              1           3             1          0
3    Monday           7              3           3             2          0
4   Tuesday           8              3           2             1          0

In [10]:
y.head()

End of the day Feeling (0 to 10)
0                                 8
1                                 4
2                                 6
3                                 5
4                                 6

In [11]:
X=pd.get_dummies(data[['Day','Sleep Hour','Exercise Hour','Study Hour','StudyDS Hour','Work Hour']],drop_first=False)

In [12]:
X.head()
#Thursday 

Sleep Hour  Exercise Hour  Study Hour  StudyDS Hour  Work Hour  Day_Friday  \
0           6              3           2             1          6           1   
1           7              2           3             2          4           0   
2           5              1           3             1          0           0   
3           7              3           3             2          0           0   
4           8              3           2             1          0           0   

   Day_Monday  Day_Saturday  Day_Sunday  Day_Thursday  Day_Tuesday  \
0           0             0           0             0            0   
1           0             1           0             0            0   
2           0             0           1             0            0   
3           1             0           0             0            0   
4           0             0           0             0            1   

   Day_Wednesday  
0              0  
1              0  
2              0  
3              0  
4              0

In [13]:
from sklearn.model_selection import train_test_split
X_train,X_test, y_train,y_test=train_test_split(X,y, test_size=.18, random_state=0)

In [14]:
from sklearn.linear_model import LinearRegression
regressor=LinearRegression()
regressor.fit(X_train,y_train)

LinearRegression()

In [15]:
y_pred=regressor.predict(X_test)

# Validation Predictions

In [16]:
regressor.predict([[6,2,1,0,0,0,0,0,0,1,0,0]])

array([[4.53463153]])

In [17]:
regressor.predict([[6,0,1,0,6,0,1,0,0,0,0,0]])

array([[6.49820103]])

In [18]:
regressor.predict([[7,1,4,1,6,0,0,0,0,0,1,0]])

array([[6.11725275]])

In [19]:
regressor.predict([[7,2,1,1,6,0,0,0,0,0,0,1]])

array([[3.73809871]])

In [20]:
regressor.predict([[6,0,1,1,6,0,0,0,0,1,0,0]])

array([[2.88095713]])

In [21]:
#R2 score
from sklearn.metrics import r2_score
print('r2-score', r2_score(y_test, y_pred))

r2-score -1.5428250996471036


In [22]:
import statsmodels.api as sm
X=sm.add_constant(X)

In [23]:
model=sm.OLS(y,X).fit()

In [24]:
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                   OLS Regression Results                                   
============================================================================================
Dep. Variable:     End of the day Feeling (0 to 10)   R-squared:                       0.255
Model:                                          OLS   Adj. R-squared:                  0.098
Method:                               Least Squares   F-statistic:                     1.619
Date:                              Tue, 22 Mar 2022   Prob (F-statistic):              0.121
Time:                                      20:35:30   Log-Likelihood:                -135.80
No. Observations:                                64   AIC:                             295.6
Df Residuals:                                    52   BIC:                             321.5
Df Model:                                        11                                         
Covariance Type:                          nonrobust                                         
=================================================================================
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
const             2.2216      2.151      1.033      0.306      -2.094       6.537
Sleep Hour        0.1041      0.306      0.340      0.735      -0.511       0.719
Exercise Hour     0.6379      0.342      1.865      0.068      -0.049       1.324
Study Hour        0.5451      0.368      1.481      0.145      -0.194       1.284
StudyDS Hour     -0.9883      0.572     -1.727      0.090      -2.137       0.160
Work Hour        -0.0292      0.118     -0.246      0.806      -0.267       0.208
Day_Friday       -0.7450      0.696     -1.071      0.289      -2.141       0.651
Day_Monday        1.0568      0.803      1.316      0.194      -0.555       2.668
Day_Saturday     -0.7685      0.902     -0.852      0.398      -2.578       1.041
Day_Sunday       -0.6998      0.999     -0.701      0.487      -2.704       1.304
Day_Thursday      0.9755      0.760      1.284      0.205      -0.549       2.500
Day_Tuesday       1.4801      0.815      1.816      0.075      -0.155       3.115
Day_Wednesday     0.9225      0.782      1.180      0.244      -0.647       2.492
==============================================================================
Omnibus:                        0.882   Durbin-Watson:                   0.847
Prob(Omnibus):                  0.643   Jarque-Bera (JB):                0.713
Skew:                          -0.256   Prob(JB):                        0.700
Kurtosis:                       2.930   Cond. No.                     7.73e+16
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is 7.25e-31. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

The r2-score from the model was -1.54. Sqince it is a negative when the value should be between 0-1, we can say that this is not a very good linear regression model with my data. Therefore meaning that there is likely no correlation between the variables.
The p-value for all of the variables in this model were greater than .05, which tells us that they are not statiscally significant. We cannot reject the null hypothesis.
